src/app/server/Server.cpp

```diff
@@ -105,6 +107,110 @@ static ::chip::PersistedCounter<chip::EventNumber> sGlobalEventIdCounter;
 static ::chip::app::CircularEventBuffer sLoggingBuffer[CHIP_NUM_EVENT_LOGGING_BUFFERS];
 #endif // CHIP_CONFIG_ENABLE_SERVER_IM_EVENT
 
+#if CHIP_CONFIG_TC_REQUIRED
+app::DefaultEnhancedSetupFlowProvider sDefaultEnhancedSetupFlowProviderInstance;
+app::EnhancedSetupFlowProvider * CommonCaseDeviceServerInitParams::sDefaultEnhancedSetupFlowProvider =
+    &sDefaultEnhancedSetupFlowProviderInstance;
+
+app::DefaultTermsAndConditionsProvider sDefaultTermsAndConditionsProviderInstance;
+app::TermsAndConditionsProvider * CommonCaseDeviceServerInitParams::sDefaultTermsAndConditionsProvider =
+    &sDefaultTermsAndConditionsProviderInstance;
```

Comment:
These should not be there, and should only be in the InitializeStaticResourcesBeforeServerInit. Otherwise someone who does override will be in a situation where there are global instances that are present in their memory image.